src/main/scala/firrtl/passes/InferWidths.scala

```diff
@@ -42,12 +42,12 @@ object InferWidths extends Pass {
     }
     def collectMinMax(w: Width): Width = w map collectMinMax match {
       case MinWidth(args) => MinWidth(unique((args.foldLeft(Seq[Width]())) {
```

Comment:
All over the place we tend to use `Seq(...)` and `Seq.empty` to create sequences when I think we should be a more explicit and call out what data structure we're instantiating. Since the appending was causing such massive problems, can you also change it to `List[Width]()` (which is what `Seq[Width]()` creates) just to be a little more obvious?

Reply:
Done. Interesting comparison of `List` vs `Vector`: https://docs.scala-lang.org/overviews/collections/performance-characteristics.html

Reply:
See also http://www.lihaoyi.com/post/BenchmarkingScalaCollections.html